packages/tools/kadena-cli/src/networks/utils/networkDisplay.ts

```diff
@@ -12,111 +14,36 @@ import type {
 import { existsSync, readFileSync } from 'fs';
 import yaml from 'js-yaml';
 import path from 'path';
-import { log } from '../../utils/logger.js';
-
-/**
- * Displays the network configuration in a formatted manner.
- *
- * @param {TNetworksCreateOptions} networkConfig - The network configuration to display.
- */
-export function displayNetworkConfig(
-  networkConfig: INetworkCreateOptions,
-): void {
-  const formatLength = 80; // Maximum width for the display
-
-  const displaySeparator = (): void => {
-    log.info(log.color.green('-'.padEnd(formatLength, '-')));
-  };
-
-  const formatConfig = (key: string, value?: string): string => {
-    const valueDisplay =
-      value !== undefined && value.trim() !== ''
-        ? log.color.green(value)
-        : log.color.red('Not Set');
-    const keyValue = `${key}: ${valueDisplay}`;
-    const remainingWidth =
-      formatLength - keyValue.length > 0 ? formatLength - keyValue.length : 0;
-    return `  ${keyValue}${' '.repeat(remainingWidth)}  `;
-  };
-
-  displaySeparator();
-  log.info(formatConfig('Network', networkConfig.network));
-  log.info(formatConfig('Network ID', networkConfig.networkId));
-  log.info(formatConfig('Network Host', networkConfig.networkHost));
-  log.info(
-    formatConfig('Network Explorer URL', networkConfig.networkExplorerUrl),
-  );
-  displaySeparator();
-}
+import type { TableHeader, TableRow } from '../../utils/tableDisplay.js';
 
 export function displayNetworksConfig(): void {
-  const formatLength = 80; // Maximum width for the display
-
-  const displaySeparator = (): void => {
-    log.info(log.color.green('-'.padEnd(formatLength, '-')));
-  };
-
-  const formatConfig = (
-    key: string,
-    value?: string,
-    isDefault?: boolean,
-  ): string => {
-    const valueDisplay =
-      (value?.trim() ?? '') !== ''
-        ? log.color.green(value!)
-        : log.color.red('Not Set');
-
-    const defaultIndicator =
-      isDefault === true ? log.color.yellow(' (Using defaults)') : '';
-    const keyValue = `${key}: ${valueDisplay}${defaultIndicator}`;
-    const remainingWidth =
-      formatLength - keyValue.length > 0 ? formatLength - keyValue.length : 0;
-    return `  ${keyValue}${' '.repeat(remainingWidth)}  `;
-  };
+  const header: TableHeader = [
+    'Network',
+    'Network ID',
+    'Network Host',
+    'Network Explorer URL',
+  ];
+  const rows: TableRow[] = [];
 
   const existingNetworks: ICustomNetworkChoice[] = getExistingNetworks();
-  const standardNetworks: string[] = ['mainnet', 'testnet'];
-
   existingNetworks.forEach(({ value }) => {
     const networkFilePath = path.join(defaultNetworksPath, `${value}.yaml`);
     const fileExists = existsSync(networkFilePath);
-    const networkConfig = fileExists
+    const networkConfig: INetworkCreateOptions = fileExists
       ? (yaml.load(
           readFileSync(networkFilePath, 'utf8'),
         ) as INetworkCreateOptions)
```

Comment:
this could also been improved instead of type casting, can be validated using zod schema